### DISCUSSION TITLE
[Feature]: Microfrontend orchestration in Remix via Dynamic Manifest injection

### DISCUSSION THREAD

**Discussion by vijaybritto** [remix-run/remix#2889] [Proposals]
### What is the new or updated feature that you are suggesting?

For microfrontends which dont use Module Federation, the asset manifest is the way to fetch assets dynamically. Since Remix is built around this idea, it would be great to have dynamic asset manifests that can injected at runtime. 

Module federation doesn't work for a lot of projects because it requires dependencies to be known at build time. Also it requires webpack 5 without which it would not work.

This is assuming that the microfrontends are split based on routes. Many apps follow this pattern since its the easiest.

Even though apps are built and served from different servers, it can still work if the manifest is shared. The browser framework can then use this to fetch the assets for the other Remix app.

Ex:

/home - First manifest
/users - second manifest
/settings - third manifest

Can this be done with Remix or is there low level primitives exposed through which we can build this ourselves?

### Why should this feature be included?

This would be a compelling reason to adopt Remix in a lot of enterprise projects (including ours).

**Comment by adrienbaron** [comment]
Hey @vijaybritto you might want to have a look at the discussion I started here: https://github.com/remix-run/remix/discussions/1989
I have a POC that does what you're talking about 😉

**Reply by vijaybritto** [reply]
Wow it's a great implementation to start with. I will reply with my thoughts after going through the poc
